packages/client/src/sync/fetcher/fetcher.ts

```diff
@@ -254,6 +273,7 @@ export abstract class Fetcher<JobTask, JobResult, StorageItem> extends Readable
       job.result = this.process(job, result)
       jobStr = this.jobStr(job, true)
       if (job.result) {
```

Comment:
not sure why the linter isn't complaining here but this should be `if (job.result !== undefined)`
```suggestion
      if (job.result !== undefined) {
```